wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -109,6 +119,32 @@ class WireJsonAdapterFactory private constructor(
       }
     }
 
+    if (Types.nextAnnotations(annotations, Uint64String::class.java) != null) {
+      when (rawType) {
+        Long::class.javaObjectType -> return UINT64_STRING_JSON_ADAPTER
+        Long::class.javaPrimitiveType -> return UINT64_STRING_JSON_ADAPTER
+        List::class.java -> {
+          if ((type as ParameterizedType).actualTypeArguments[0] == Long::class.javaObjectType) {
+            return LIST_OF_UINT64_STRING_JSON_ADAPTER
+          }
+        }
+        Map::class.java -> TODO()
```

Comment:
YES. Yikes.

Reply:
I assume these values aren’t omitted, but we do need to change how they’re encoded?

Reply:
Yes they do.